fuzz_lightyear/datastore.py

```diff
@@ -166,53 +173,70 @@ def inject_user_defined_variables(func: Callable) -> Callable:
+                    raise TypeError
+
+                if operation_id is not None:
+                    value = mapping[arg_name][operation_id]()
+
+                else:
+                    from .supplements.factory import returns_none
+                    fixture_func = mapping[arg_name].default_factory()
+                    if fixture_func == returns_none:
+                        raise TypeError
+                    value = fixture_func()
+                if (
+                    arg_name in type_annotations
+                    and not isinstance(type_annotations[arg_name], type(List))
+                ):
+                    # If type annotations are used, use that to cast
+                    # values for input.
+                    value = type_annotations[arg_name](value)
+
+                kwargs[arg_name] = value
+
+            if getattr(func, '_fuzz_cache', None) is not None:
+                func._fuzz_cache[arg_tuple] = func(*args, **kwargs)  # type: ignore
+            else:
+                func._fuzz_cache = {arg_tuple: func(*args, **kwargs)}  # type: ignore
             return func._fuzz_cache[arg_tuple]  # type: ignore
 
-        expected_args = _get_injectable_variables(func)
-        type_annotations = inspect.getfullargspec(func).annotations
-
-        for index, arg_name in enumerate(expected_args):
-            if index < len(args):
-                # This handles the case of explicitly supplied
-                # positional arguments, so that we don't pass func
-                # two values for the same argument.
-                continue
-
-            if arg_name not in mapping:
-                raise TypeError
-
-            # We don't allow specific endpoints here, and only use the default.
-            # If we don't have a default value, then we raise an error.
-            from .supplements.factory import returns_none
-            fixture_func = mapping[arg_name].default_factory()
-            if fixture_func == returns_none:
-                raise TypeError
-            value = fixture_func()
-            if (
-                arg_name in type_annotations
-                and not isinstance(type_annotations[arg_name], type(List))
-            ):
-                # If type annotations are used, use that to cast
-                # values for input.
-                value = type_annotations[arg_name](value)
-
-            kwargs[arg_name] = value
-
-        if getattr(func, '_fuzz_cache', None) is not None:
-            func._fuzz_cache[arg_tuple] = func(*args, **kwargs)  # type: ignore
-        else:
-            func._fuzz_cache = {arg_tuple: func(*args, **kwargs)}  # type: ignore
-        return func._fuzz_cache[arg_tuple]  # type: ignore
-
-    return wrapped
+        return wrapped
+    if original_callable and callable(original_callable):
+        return inject_variables(original_callable)
+    elif callable(original_callable):
+        raise TypeError  # This function was called without its keyword
```

Comment:
I'm actually super confused about how code path is hit. In what situation would the following statement be true?

```
not bool(original_callable) and callable(original_callable)
```